airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -91,6 +99,61 @@ def get_container_termination_message(pod: V1Pod, container_name: str):
         return container_status.state.terminated.message if container_status else None
 
 
+class PodLogsConsumer:
+    """
+    PodLogsConsumer is responsible for pulling pod logs from a stream with checking a container status before
+    reading data.
+    This class is a workaround for the issue https://github.com/apache/airflow/issues/23497
+    """
+
+    def __init__(
+        self,
+        response: HTTPResponse,
+        pod: V1Pod,
+        pod_manager: PodManager,
+        container_name: str,
+        timeout: int = 120,
```

Comment:
maybe it's better named ... `post_completion_timeout` or `completion_timeout` or `termination_timeout`.  when i think of timeout i think of max time beteen each chunk.

Reply:
done